lib/modules.nix

```diff
@@ -266,6 +266,16 @@ rec {
               turned off.
             '';
           };
+
+          _module.specialArgs = mkOption {
```

Comment:
I don't see this being used in the rest of this PR, is this necessary?

Reply:
It sets the appropriate option flags, such as `readOnly`.
It sets the value as well, in `default`. Not sure if we want the default for docs. Maybe not actually. Will move that definition to `config`.